src/services/redisSessionStorage.ts

```diff
@@ -862,24 +870,23 @@ export default class RedisSessionStorage extends RedisStorageUtils
       this.redisClient.get(`${prefix}${token}`, (err, value) => {
         if (err) {
           // Client returns an error.
-          return resolve(left<Error, User>(err));
+          return resolve(E.left(err));
         }
 
         if (value === null) {
-          return resolve(left<Error, User>(sessionNotFoundError));
+          return resolve(E.left(sessionNotFoundError));
         }
 
         this.loadSessionBySessionToken(value as SessionToken).then(
-          (errorOrSession: Either<Error, User>) => {
-            errorOrSession.fold(
-              error => resolve(left<Error, User>(error)),
-              session => {
-                resolve(right<Error, User>(session));
-              }
-            );
-          },
+          flow(
+            E.mapLeft(error => resolve(E.left(error))),
+            E.map(session => {
+              resolve(E.right(session));
+            }),
+            E.toUnion
```

Comment:
this should be unnecessary

Reply:
you are rigth, thanks!

Reply:
I had to revert this changes: the unit tests will stuck on it.